backend/executor.go

```diff
@@ -181,9 +189,27 @@ func (executor *grpcExecutor) ExecuteActivity(ctx context.Context, iid api.Insta
 }
 
 // Shutdown implements Executor
-func (g *grpcExecutor) Shutdown(ctx context.Context) {
+func (g *grpcExecutor) Shutdown(ctx context.Context) error {
 	// closing the work item queue is a signal for shutdown
 	close(g.workItemQueue)
+
+	// Iterate through all pending items and close them to unblock the goroutines waiting on this
+	g.pendingActivities.Range(func(_, value any) bool {
+		p, ok := value.(*activityExecutionResult)
+		if ok {
+			close(p.complete)
```

Comment:
Shouldn't we check if the `complete` channel is already closed as it's closed at multiple places.

Reply:
If the channel is in the map, it's not been closed. Before closing the channel, the calls use `LoadAndDelete` which means that the channel is "removed from the map" before `close` is called on it